geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/utils/ConfluentMetadata.scala

```diff
@@ -0,0 +1,129 @@
+import org.opengis.feature.simple.SimpleFeatureType
+
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+class ConfluentMetadata(val schemaRegistry: SchemaRegistryClient) extends GeoMesaMetadata[String] with LazyLogging {
+
+  private val topicSftCache: LoadingCache[String, String] =
+    Caffeine.newBuilder()
+      .expireAfterWrite(10, TimeUnit.MINUTES)
+      .build(
+        new CacheLoader[String, String] {
+          def load(topic: String): String = getSftSpecForTopic(topic).orNull
+        }
+      )
+
+  protected def getSftSpecForTopic(topic: String): Option[String] =
+    try {
+      val subject = topic + SUBJECT_POSTFIX
+      val schemaId = schemaRegistry.getLatestSchemaMetadata(subject).getId
+      val sft = schemaToSft(schemaRegistry.getByID(schemaId), topic) // todo: any restrictions on sftName not on topic?
+      KafkaDataStore.setTopic(sft, topic)
+      Option(SimpleFeatureTypes.encodeType(sft, includeUserData = true))
+    } catch {
+      case NonFatal(t) =>
+        logger.error("Error retrieving schema from confluent registry.  Returning None.", t)
+        None
+    }
+
+  override def getFeatureTypes: Array[String] = schemaRegistry.getAllSubjects.asScala
+                                                              .filter(_.endsWith(SUBJECT_POSTFIX))
+                                                              .map(s => s.substring(0, s.lastIndexOf(SUBJECT_POSTFIX)))
+                                                              .toArray
+
+  override def read(typeName: String, key: String, cache: Boolean): Option[String] = {
+    if (key != GeoMesaMetadata.ATTRIBUTES_KEY) {
+      logger.warn(s"Requested read on ConfluentMetadata with unsupported key $key. " +
+        s"ConfluentMetadata only supports ${GeoMesaMetadata.ATTRIBUTES_KEY}")
+      None
+    } else {
+      if (!cache) {
+        getSftSpecForTopic(typeName)
+      } else {
+        Option(topicSftCache.get(typeName))
+      }
+    }
+  }
+
+  override def invalidateCache(typeName: String, key: String): Unit = {
+    if (key != GeoMesaMetadata.ATTRIBUTES_KEY) {
+      logger.warn(s"Requested invalidate cache on ConfluentMetadata with unsupported key $key. " +
+        s"ConfluentMetadata only supports ${GeoMesaMetadata.ATTRIBUTES_KEY}")
+    } else {
+      topicSftCache.invalidate(typeName)
+    }
+  }
+
+  override def close(): Unit = {}
+
+  override def scan(typeName: String, prefix: String, cache: Boolean): Seq[(String, String)] =
+    throw new NotImplementedError(s"ConfluentMetadata only supports ATTRIBUTES_KEY")
+
+  override def insert(typeName: String, key: String, value: String): Unit = {}
+  override def insert(typeName: String, kvPairs: Map[String, String]): Unit = {}
+  override def remove(typeName: String, key: String): Unit = {}
+  override def delete(typeName: String): Unit = {}
+}
+
+object ConfluentMetadata extends LazyLogging {
+  val SUBJECT_POSTFIX = "-value"
```

Comment:
Could you add a comment here on why we are looking for just topics ending in '-value'? Is that a confluent standard?

Reply:
Yeah, i'll definitely add a comment.  Before I do, do you know if there is a need to support more?  It's default behavior of the serializer/deserializer, also how ksql registers all schemas it generates.
For my use-cases it's sufficient.

Reply:
I'll probably just add in the comment that we can support more in the future if needed

Reply:
I don't know - I think if it supports your use case then that is good enough for now